src/tunacode/tools/read_file.py

```
@@ -73,61 +85,117 @@ def _read_sync(
+    signal: asyncio.Event | None,
+    on_update: AgentToolUpdateCallback,
+) -> AgentToolResult:
+    _ = (tool_call_id, on_update)
+    if signal is not None and signal.is_set():
+        raise UserAbortError("Tool execution aborted: read_file")
+
+    filepath = args.get("filepath")
+    offset = args.get("offset", 0)
+    limit = args.get("limit")
+    if not isinstance(filepath, str):
+        raise ToolRetryError("Invalid arguments for tool 'read_file': 'filepath' must be a string.")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)